[ASTGen] Introduce nullable variants of bridging wrappers

hamishknight opened the pull request:
For nullable nodes, introduce both a non-null and nullable variant of the bridging wrapper. This allows us to annotate the necessary parameters as nullable, but keep the returns of the bridged `createParsed` methods non-null.

hamishknight commented:
@swift-ci please test

ahoppen reviewed (approved):
I like this design but I’d like to hear @rintaro’s and @bnbarham’s thoughts on this as well.

hamishknight committed:
[ASTGen] Introduce nullable variants of bridging wrappers

For nullable nodes, introduce both a non-null and
nullable variant of the bridging wrapper. This
allows us to annotate the necessary parameters
as nullable, but keep the returns of the bridged
`createParsed` methods non-null.

bnbarham reviewed (approved):
The `asNullable` is a little unfortunate, but I'm in favor of this given all the places that don't allow nullable but took them anyway.
